### PR TITLE
Atp upload updates

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

File: interface/resources/qml/AssetServer.qml [hunk 1]
```diff
@@ -220,47 +219,67 @@ Window {
...
+                function(err, path) {
+                    print(err, path);
+                    if (!err) {
+                        uploadProgressLabel.text = "Upload Complete";
+                        timer.interval = 1000;
+                        timer.repeat = false;
+                        timer.triggered.connect(function() {
+                            uploadSpinner.visible = false;
+                            uploadButton.enabled = true;
+                            uploadOpen = false;
```
Comment (huffman) [inline interface/resources/qml/AssetServer.qml:287]: bit of DRY here, I'm going to try changing this around a bit tomorrow.

File: interface/src/scripting/AssetMappingsScriptingInterface.cpp [hunk 2]
```diff
@@ -59,11 +59,16 @@ void AssetMappingsScriptingInterface::getMapping(QString path, QJSValue callback
     request->start();
 }
 
-void AssetMappingsScriptingInterface::uploadFile(QString path, QString mapping, QJSValue callback) {
+void AssetMappingsScriptingInterface::uploadFile(QString path, QString mapping, QJSValue startedCallback, QJSValue completedCallback) {
+    static const QString helpText =
+        "Upload your asset to a specific folder by enter the full path. Specifying "
```
Comment (birarda) [inline interface/src/scripting/AssetMappingsScriptingInterface.cpp]: entering the full path

File: interface/src/scripting/AssetMappingsScriptingInterface.cpp [hunk 3]
```diff
@@ -79,19 +84,22 @@ void AssetMappingsScriptingInterface::uploadFile(QString path, QString mapping,
         auto button = offscreenUi->messageBox(OffscreenUi::ICON_QUESTION, "Overwrite File", message,
                                               QMessageBox::Yes | QMessageBox::No, QMessageBox::No);
         if (button == QMessageBox::No) {
+            completedCallback.call({ -1 });
```
Comment (Atlante45) [inline interface/src/scripting/AssetMappingsScriptingInterface.cpp:87]: Need to check it's callable

File: interface/src/scripting/AssetMappingsScriptingInterface.cpp [hunk 4]
```diff
@@ -59,11 +59,16 @@ void AssetMappingsScriptingInterface::getMapping(QString path, QJSValue callback
...
+    static const QString helpText =
+        "Upload your asset to a specific folder by enter the full path. Specifying "
+        "a new folder name will automatically create that folder for you.";
+
     auto offscreenUi = DependencyManager::get<OffscreenUi>();
-    auto result = offscreenUi->inputDialog(OffscreenUi::ICON_NONE, "Enter asset path:", "", mapping);
+    auto result = offscreenUi->inputDialog(OffscreenUi::ICON_INFORMATION, "Specify Asset Path", helpText, mapping);
 
     if (!result.isValid()) {
+        completedCallback.call({ -1 });
```
Comment (Atlante45) [inline interface/src/scripting/AssetMappingsScriptingInterface.cpp:71]: Need to check it's callable

File: interface/src/scripting/AssetMappingsScriptingInterface.cpp [hunk 5]
```diff
@@ -79,19 +84,22 @@ void AssetMappingsScriptingInterface::uploadFile(QString path, QString mapping,
         auto button = offscreenUi->messageBox(OffscreenUi::ICON_QUESTION, "Overwrite File", message,
                                               QMessageBox::Yes | QMessageBox::No, QMessageBox::No);
         if (button == QMessageBox::No) {
+            completedCallback.call({ -1 });
             return;
         }
     }
 
+    startedCallback.call();
```
Comment (Atlante45) [inline interface/src/scripting/AssetMappingsScriptingInterface.cpp:92]: Need to check it's callable

File: interface/src/scripting/AssetMappingsScriptingInterface.cpp (hunk 4 above)
Comment (huffman) [inline interface/src/scripting/AssetMappingsScriptingInterface.cpp:71]: Why? The docs say it will do nothing if it's not callable.

Comment (Atlante45) [inline interface/src/scripting/AssetMappingsScriptingInterface.cpp:71]: Oh ok, I didn't know that. Nice!

**Pull request merged**